src/coreclr/src/jit/hwintrinsicxarch.cpp

```diff
@@ -1173,19 +1173,209 @@ GenTree* Compiler::impBaseIntrinsic(NamedIntrinsic        intrinsic,
 
 GenTree* Compiler::impSSEIntrinsic(NamedIntrinsic intrinsic, CORINFO_METHOD_HANDLE method, CORINFO_SIG_INFO* sig)
 {
-    GenTree* retNode  = nullptr;
-    GenTree* op1      = nullptr;
-    GenTree* op2      = nullptr;
-    GenTree* op3      = nullptr;
-    GenTree* op4      = nullptr;
-    int      simdSize = HWIntrinsicInfo::lookupSimdSize(this, intrinsic, sig);
+    GenTree*  retNode  = nullptr;
+    GenTree*  op1      = nullptr;
+    GenTree*  op2      = nullptr;
+    int       simdSize = HWIntrinsicInfo::lookupSimdSize(this, intrinsic, sig);
+    var_types baseType = TYP_UNKNOWN;
 
     // The Prefetch and StoreFence intrinsics don't take any SIMD operands
     // and have a simdSize of 0
     assert((simdSize == 16) || (simdSize == 0));
 
     switch (intrinsic)
     {
+        case NI_SSE_CompareGreaterThan:
+        {
+            assert(sig->numArgs == 2);
+            op2      = impSIMDPopStack(TYP_SIMD16);
+            op1      = impSIMDPopStack(TYP_SIMD16);
+            baseType = getBaseTypeOfSIMDType(sig->retTypeSigClass);
+            assert(baseType == TYP_FLOAT);
+
+            if (compExactlyDependsOn(InstructionSet_AVX))
```

Comment:
@davidwrighton, should I be using `Exactly` or `Opportunistically` here?

Based on https://github.com/dotnet/runtime/blob/master/docs/design/coreclr/botr/vectors-and-intrinsics.md#mechanisms-in-the-jit-to-generate-correct-code-to-handle-varied-instruction-set-support, it sounds like I want `Opportunistically` since the two paths are semantically equivalent and the AVX path is just an optimization for better codegen. Is that correct?

Reply:
Yes. My analysis matches yours. These can (and should) be compared using `Opportunistically`.

Reply:
Thanks! I just asked a related question on https://github.com/dotnet/runtime/pull/33924#issuecomment-613015698, since I think it falls into the same category.

Reply:
Switched to using `compOpportunisticallyDependsOn` as both paths are TYP_SIMD16 and are semantically equivalent.